py3status/modules/xrandr_toggle.py

```diff
@@ -0,0 +1,160 @@
+    def _xrandr(self):
+            return check_output(['xrandr']).split('\n')
+
+    # return connected monitor using xrandr and number for index
+    def get_status(self, i3s_output_list, i3s_config):
+        try:
+            # monitor number (index) must be bigger than 0
+            number = int(self.number) - 1
+            if number < 0:
+                return self._error_response(i3s_config['color_bad'],
+                                            'invalid number')
+            monitors = []
+            # width x height + left + top
+            regex = r'\d{1,}x\d{1,}\+\d{1,}\+\d{1,}'
+            # for each line find (dis)connected monitors
+            for line in self.status:
+                if 'connected' in line:
+                    details = line.split(' ')
+                    if len(details) > 2:
+                        source = {}
+                        source['id'] = details[0]
+                        if details[1] == 'disconnected':
+                            source['connected'] = 0
+                            source['state'] = 'off'
+                            source['status'] = self.format_disconnected
+                        else:
+                            source['connected'] = 1
+                            if search(regex, details[2]):
+                                source['state'] = 'auto'
+                                source['status'] = self.format_on
+                            else:
+                                source['state'] = 'off'
+                                source['status'] = self.format_off
+                        monitors.append(source)
+
+            # index bigger than array of monitors found
+            if number >= len(monitors):
+                return self._error_response(i3s_config['color_bad'],
+                                            'number is too high')
+            # if source is set, use that instead of number
+            if len(self.source) > 0:
+                self.mon = next((x for x in monitors
+                                if x['id'] == self.source),
+                                None)
+                if self.mon is None:
+                    return self._error_response(i3s_config['color_bad'],
+                                                'monitor not found')
+            else:
+                self.mon = monitors[number]
+
+            if self.mon['connected'] == 0:
+                color = i3s_config['color_bad']
+            elif self.mon['state'] == 'auto':
+                color = i3s_config['color_good']
+            else:
+                color = i3s_config['color_degraded']
+
+            response = {
+                'cached_until': time() + self.cache_timeout,
+                'color': color,
+                'full_text': self.format.format(screen=self.mon['id'],
+                                                status=self.mon['status'])
+            }
+            return response
+        except:
```

Comment:
Raw exceptions are horrible things.  If things are breaking it's nice to find out and fix them.

If there is an known issue then just catch that.